### ISSUE TITLE
Memetic: Gradient-informed quantum adjustment

### ISSUE THREAD

**Issue by stservice** [stSoftwareAU/NEAT-TS#1325]
## Parent Issue
This is a sub-issue of #1321 (Plan to improve memetic evolution)

## Summary
The current quantum adjustment uses uniform random scaling. This could be improved by leveraging information from the elastic backpropagation system to inform the direction and magnitude of adjustments.

## Current Implementation
In `src/blackbox/FineTune.ts:21-51`:
```typescript
export function quantumAdjust(
  currentBest: number,
  previousBest: number,
  forwardOnly: boolean,
  backtrack: boolean,
): { value: number; changed: boolean } {
  const diff = currentBest - previousBest;
  // ... uses random scaling
  delta = diff * Math.random() * scale;  // Pure random, no gradient info
}
```

## Proposed Improvement
- During fine-tuning, optionally run a single backpropagation pass to gather gradient information
- Use the gradient direction to bias quantum adjustments
- Implement a hybrid approach: gradient-informed direction + random magnitude
- Allow configuration between pure-random (current) and gradient-informed modes

## Technical Approach
1. Add optional gradient collection during fine-tuning
2. Weight the random adjustment direction by gradient sign
3. Scale adjustment magnitude based on gradient magnitude
4. Preserve the quantum discretisation for stability

## Expected Benefits
- More targeted fine-tuning that moves in informed directions
- Potentially faster convergence
- Maintains the exploration benefits of randomness while adding gradient guidance

## Requirements
- [ ] Follow TDD - write failing tests first
- [ ] Follow DRY principles - avoid code duplication
- [ ] Unit tests must verify correct behaviour (not performance)
- [ ] If performance improvement is claimed, include benchmark results as evidence
- [ ] Use Australian English spelling (e.g., behaviour, optimise)

**Comment by stservice** [comment]
## Automated Processing Failed (First Attempt)

The automated worker was unable to complete this issue on the first attempt.

### Failure Details
Claude timed out after 3600 seconds (60 minutes).

This issue may be too complex, too large, or require additional permissions that Claude doesn't have.

<details>
<summary>Last output from Claude (click to expand)</summary>

```
No output captured
```
</details>

### What happens next?
- This issue has been marked with the `failed-once` label
- The worker will **automatically retry** this issue on the next scan
- If it fails again, the issue will be marked as `failed` and require manual intervention

### To prevent automatic retry
If you want to work on this manually, remove the `failed-once` label and assign yourself.

**Comment by stservice** [comment]
## Automated Processing Failed (Second Attempt - Permanently Failed)

The automated worker has failed to complete this issue twice and will not retry automatically.

### Failure Details
Claude timed out after 3600 seconds (60 minutes).

This issue may be too complex, too large, or require additional permissions that Claude doesn't have.

<details>
<summary>Last output from Claude (click to expand)</summary>

```
No output captured
```
</details>

### What happened?
- This issue failed on the first attempt and was marked with `failed-once`
- The worker retried and failed again on the second attempt
- The issue has been marked with the `failed` label to prevent repeated attempts
- Manual intervention is required

### Why this might be happening
- The issue may be too complex or large for automated processing
- The issue description may need more detail or clarification
- There may be dependencies or permissions that Claude doesn't have

### To retry this issue
1. Investigate and address the underlying problem (check the failure details above)
2. Consider simplifying or breaking down the issue into smaller tasks
3. Remove the `failed` label from this issue
4. The worker will pick it up again on the next scan

**Comment by stservice** [comment]
## Closing as substantially implemented via trajectory momentum

Audit of open issues (#1351) found that the goal of gradient-informed quantum adjustment has been achieved through a different but effective approach:

- **Adaptive step sizing** via `calculateEffectiveStep()` in `FineTune.ts`: Uses error magnitude normalisation and bounded step scaling.
- **Trajectory momentum** (lines 99-126, 287-300, 338-350): When momentum is available from ancestry tracking, it biases adjustments in the direction of consistent improvement (factor 0.5-2.0).
- **Configurable quantum step size** (#1330, recently merged): Step sizing adapts based on training progress.
- **Bias-weight coordination** (#1331, recently merged): Coordinates quantum adjustments between biases and weights.

While the approach differs from classical gradient computation (trajectory momentum vs backpropagation gradients), the outcome is the same: quantum adjustments are informed by the direction and magnitude of recent improvements. This issue also failed automated processing twice, suggesting the original approach of adding actual gradient computation may not be the right fit for the architecture.
